services/rfq/relayer/service/handlers.go

```diff
@@ -186,6 +186,22 @@ func (q *QuoteRequestHandler) handleSeen(ctx context.Context, span trace.Span, r
 		return nil
 	}
 
+	latestBlock := q.Origin.LatestBlock()
+
+	canRelay, err := q.canRelayBasedOnVolumeAndConfirmations(request, latestBlock, q.volumeLimit)
+	if err != nil {
+		span.AddEvent("could not determine if can relay")
+		return fmt.Errorf("could not determine if can relay: %w", err)
+	}
+	if !canRelay {
+		err = q.db.UpdateQuoteRequestStatus(ctx, request.TransactionID, reldb.CommittedPending, &request.Status)
```

Comment:
IMO we shouldn't update to CommittedPending here because we don't want to allow the balance to be consumed by another request until this one is ready to relay

Reply:
at this point we de facto committed confirmed this, we just cant act on it yet? it follow similar order to 

```
func (q *QuoteRequestHandler) handleCommitPending(ctx context.Context, span trace.Span, request reldb.QuoteRequest) (err error) {
	earliestConfirmBlock := request.BlockNumber + q.Origin.Confirmations

	latestBlock := q.Origin.LatestBlock()
	shouldContinue := latestBlock >= earliestConfirmBlock
	span.AddEvent("pending_check", trace.WithAttributes(
		attribute.Int("latest_block", int(latestBlock)),
		attribute.Int("earliest_confirm_block", int(earliestConfirmBlock)),
		attribute.Bool("should_continue", shouldContinue),
	))
	...
```